src/shims/unix/linux/eventfd.rs

```diff
@@ -62,23 +62,27 @@ impl FileDescription for Event {
         &self,
         self_ref: &FileDescriptionRef,
         _communicate_allowed: bool,
-        bytes: &mut [u8],
+        _bytes: &mut [u8],
+        len: usize,
         ptr: Pointer,
         dest: &MPlaceTy<'tcx>,
         ecx: &mut MiriInterpCx<'tcx>,
     ) -> InterpResult<'tcx> {
+        // eventfd read at the size of u64.
+        let buf_place = ecx.ptr_to_mplace(ptr, ecx.machine.layouts.u64);
```

Comment:
This is a fail attempt, it compiles but the test threw UB error: 
```rust
error: Undefined Behavior: accessing memory based on pointer with alignment 2, but alignment 8 is required
  --> tests/pass-dep/libc/libc-eventfd.rs:13:29
   |
LL |     let res: i32 = unsafe { libc::read(fd, buf.as_mut_ptr().cast(), N).try_into().unwrap() };
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ accessing memory based on pointer with alignment 2, but alignment 8 is required
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE:
   = note: inside `read_bytes::<8>` at tests/pass-dep/libc/libc-eventfd.rs:13:29: 13:71
note: inside `test_read_write`
  --> tests/pass-dep/libc/libc-eventfd.rs:33:15
   |
LL |     let res = read_bytes(fd, &mut buf);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `main`
  --> tests/pass-dep/libc/libc-eventfd.rs:8:5
   |
LL |     test_read_write();
   |     ^^^^^^^^^^^^^^^^^
```
Writing to a slice that at least has size 8 should be different from writing to u64.

Reply:
Yea, alignment was never required here. Places can also be unaligned I think. Or we read unaligned from the place. I don't remember which it was but you'll need to somehow do an unaligned read

Reply:
Ah, yes. We'll either need a `TyAndLayout` for a packed struct containing a `u64` field, or we need to add a `ptr_to_mplace_unaligned` on the rustc side.